purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1114,32 +1120,17 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         completion: ReceiveCustomerInfoCallback? = null
     ) {
         state.appInBackground.let { appInBackground ->
-            fetchAndCacheCustomerInfo(appUserID, appInBackground, completion)
+            customerInfoRetriever.retrieveCustomerInfo(
+                appUserID,
+                CacheFetchPolicy.FETCH_CURRENT,
+                appInBackground,
+                this::cacheCustomerInfoAndSendToDelegateIfChanges,
+                completion
+            )
             fetchAndCacheOfferings(appUserID, appInBackground)
         }
     }
 
-    private fun fetchAndCacheCustomerInfo(
-        appUserID: String,
-        appInBackground: Boolean,
-        completion: ReceiveCustomerInfoCallback? = null
-    ) {
-        deviceCache.setCustomerInfoCacheTimestampToNow(appUserID)
-        backend.getCustomerInfo(
-            appUserID,
-            appInBackground,
-            { info ->
-                cacheCustomerInfo(info)
-                sendUpdatedCustomerInfoToDelegateIfChanged(info)
```

Comment:
These 2 are the operations I mentioned on this other comment: https://github.com/RevenueCat/purchases-android/pull/546/files#r896692163